src/polysolve/nonlinear/Solver.cpp

```diff
@@ -195,6 +195,7 @@ namespace polysolve::nonlinear
             this->m_stop.fDelta, this->m_stop.gradNorm, this->m_stop.xDelta);
 
         update_solver_info(objFunc.value(x));
+        objFunc.step_accepted(this->m_current.iterations, x);
```

Comment:
then, can it be implemented inside solution changed?

Reply:
or post step?

Reply:
This cannot be put inside solution changed since it gets called a lot in the line search and we want something once per iteration. Post step could work, but we need something that also gets called after the first solution changed. So we'd either need some function like this (we can change the name) or add a post_step function call between lines 182 and 198.